pkg/dataobj/internal/dataset/page_reader.go

```diff
@@ -193,20 +193,30 @@ func (pr *pageReader) init(ctx context.Context) error {
 		return fmt.Errorf("opening page for reading: %w", err)
 	}
 
+	if pr.presenceReader == nil {
+		pr.presenceReader = bufio.NewReader(presenceReader)
+	} else {
+		pr.presenceReader.Reset(presenceReader)
+	}
 	if pr.presenceDec == nil {
-		pr.presenceDec = newBitmapDecoder(bufio.NewReader(presenceReader))
+		pr.presenceDec = newBitmapDecoder(pr.presenceReader)
 	} else {
-		pr.presenceDec.Reset(bufio.NewReader(presenceReader))
+		pr.presenceDec.Reset(pr.presenceReader)
 	}
 
+	if pr.valuesReader == nil {
+		pr.valuesReader = bufio.NewReader(valuesReader)
+	} else {
+		pr.valuesReader.Reset(valuesReader)
+	}
```

Comment:
nit: The "make or reset" checks are starting to be a little noisy; should we have `pr.getPresenceReader()` etc to make this method easier to follow?